api/swagger.yml

```diff
@@ -1632,8 +1632,47 @@ components:
       required:
         - installation_id
         - reports
-
-
+    ExternalPrincipalList: 
+      type: object
+      required:
+        - pagination
+        - results
+      properties:
+        pagination:
+          $ref: "#/components/schemas/Pagination"
+        results:
+          type: array
+          items:
+            $ref: "#/components/schemas/ExternalPrincipal"
+    ExternalPrincipalSettings:
+      type: object
+      additionalProperties:
+        type: string
+      description: Additional settings to be consumed by the remote authenticator
```

Comment:
I don't understand what this is. Is it Opaque? Can you add an example?

Reply:
This is opaque on purpose to avoid breaking the API. 
An example is forcing session_name or only matching based on role_name in IAM. 
Another example would be specifying max ttl.